### PR TITLE
Improve mobile experience of Profile Page and App Sidebar/Header

### PR DESCRIPTION
**Pull request by kmshin1397**
In this PR:
- Improve the mobile experience of the Profile Page
  - Expand the length of the email and phone number fields in the preferences form on small screens (since half of a phone screen is too small to generally display full emails)
  - Refactor the tokens list to use MUIDataTable to take advantage of its built-in responsive display
- Improve the mobile experience of the app's sidebar and header
  - Do not display the logos on small screens to save space in the header (Closes issue #1156)
  - Use the `temporary` variant of the `Drawer` component for the sidebar on mobile, so that it appears above the content instead of squishing it to the side (which on mobile means like 200px for the entire contents of a page).
- Remove `Main.css` and move those styles into the relevant JSX components with Material-UI `makeStyles()`

sidebar

### CODE REVIEW

**Comment by kmshin1397** [comment]
@dmitryduev Good idea. Done:
Screenshot 2021-02-09 135347

**Review by dmitryduev — APPROVED** [review]
👍🏻

**Comment by profjsb** [comment]
Some formatting issues on tables:
image
image

Perhaps need some font size changes?

File: static/js/components/HeaderContent.jsx.template [hunk 1]
```diff
@@ -2,31 +2,67 @@ import React from "react";
...
+      borderRight: "none",
+      paddingRight: "0",
+    },
+  },
+}));
+const HeaderContent = () => {
+  const classes = useStyles();
+  return (
+    <div className={classes.topBannerContent}>
+      <div style= {%- raw -%} {{display: "inline-flex", flexDirection: "row" }} {%- endraw -%} >
```
Comment (stefanv) [inline static/js/components/HeaderContent.jsx.template:48]: I think

```
<div style="display: inline-flex, flexDirection: row">
```

should work just as well.

File: static/js/components/TokenList.jsx [hunk 2]
```diff
@@ -4,22 +4,63 @@ import { useDispatch } from "react-redux";
...
+            // Cancel out small screen styling and replace
+            padding: "0px",
+            paddingRight: "2px",
+            flexFlow: "row nowrap",
+          },
+        },
+        selectRoot: {
+          marginRight: "0.5rem",
+          [theme.breakpoints.up("sm")]: {
+            marginLeft: "0.rem",
```
Comment (stefanv) [inline static/js/components/TokenList.jsx:47]:
```suggestion
            marginLeft: "0",
```

File: static/js/components/HeaderContent.jsx.template [hunk 3]
```diff
@@ -2,31 +2,67 @@ import React from "react";
...
-  <div className={styles.topBannerContent}>
-    <div style= {%- raw -%} {{display: "inline-flex", flexDirection: "row" }} {%- endraw -%} >
-      <div className={styles.logos}>
-        {%- for logo in app.logos %}
-        <div className={styles.logoContainer}>
-          <Logo src="{{ logo.src }}" altText="{{ logo.alt_text }}" />
+const useStyles = makeStyles((theme) => ({
+  topBannerContent: {
+    position: "relative",
+    paddingLeft: "1em",
```
Comment (stefanv) [inline static/js/components/HeaderContent.jsx.template:13]: rem?

**Review by stefanv — APPROVED** [review]

**Review by profjsb — CHANGES_REQUESTED** [review]
Looking good! Some small formatting issues noted in the comments. Note I'm seeing two console Warnings on the front page:
```
Warning: Failed prop type: Invalid prop `initialValues.categories` of type `object` supplied to `WidgetPrefsDialog`, expected `string`.
    in WidgetPrefsDialog (created by NewsFeed)
    in NewsFeed (created by HomePage)
...
NewsFeed NewsFeed.jsx:146
```
and 
```
Warning: Failed prop type: Invalid prop `value` of type `array` supplied to `CustomGroupsWidget`, expected `string`.
    in CustomGroupsWidget
    in Unknown (created by ArrayField)
...
```

File: static/js/components/TokenList.jsx (hunk 2 above)
Comment (kmshin1397) [reply static/js/components/TokenList.jsx:47]: fixed

File: static/js/components/HeaderContent.jsx.template (hunk 3 above)
Comment (kmshin1397) [reply static/js/components/HeaderContent.jsx.template:13]: fixed

File: static/js/components/HeaderContent.jsx.template (hunk 1 above)
Comment (kmshin1397) [reply static/js/components/HeaderContent.jsx.template:48]: I think it's like this because ESLint doesn't like the passing strings into the style thing.

**Comment by kmshin1397** [comment]
> Looking good! Some small formatting issues noted in the comments. Note I'm seeing two console Warnings on the front page:
> 
> ```
> Warning: Failed prop type: Invalid prop `initialValues.categories` of type `object` supplied to `WidgetPrefsDialog`, expected `string`.
>     in WidgetPrefsDialog (created by NewsFeed)
>     in NewsFeed (created by HomePage)
> ...
> NewsFeed NewsFeed.jsx:146
> ```
> 
> and
> 
> ```
> Warning: Failed prop type: Invalid prop `value` of type `array` supplied to `CustomGroupsWidget`, expected `string`.
>     in CustomGroupsWidget
>     in Unknown (created by ArrayField)
> ...
> ```

Thanks for the detailed feedback, @profjsb! I've actually started working on another PR for the tables stuff and I believe the console errors are fixed in there as well. That just kind of fell to the wayside with the ZTF matchfile work this week, but I can probably get back to it in the next few days. So I think I would prefer to merge this PR as is and address your comments in the next PR. I didn't want to make these too big by touching every component I found that needed these fixes.

**Comment by stefanv** [comment]
Go ahead and merge on green.  Thanks, @kmshin1397!

**Comment by kmshin1397** [comment]
Only failure is weather widget (unrelated)

**Pull request merged**